src/main/protobuf/spawn.proto

```diff
@@ -356,8 +390,10 @@ message ExecLogEntry {
     string mnemonic = 4;
   }
 
-  // The entry ID. Must be nonzero.
-  int32 id = 1;
+  // If nonzero, then this entry may be referenced by later entries by this ID.
+  // Nonzero IDs must be strictly monotonically increasing within a log and
+  // should be consecutive to simplify bookkeeping for consumers.
+  uint32 id = 1;
```

Comment:
Saving memory in the reconstructor is a great point - you've convinced me of the merit of (1). For (2) I'm okay with a monotonic implementation, I'd just rather not make a promise at this time (although it might very well not matter much due to Hyrum's Law).